WordPress/gutenberg
# Add template replace flow to template inspector

## What?
Add template replace flow to template inspector - fixes https://github.com/WordPress/gutenberg/issues/54562

## Why?
Twenty Twenty-Four is exploring adding alternate template types for the home, index, single and page templates. While the effort in https://github.com/WordPress/gutenberg/pull/51477 brings alternate template selection to pages, templates still do not have this capability.

There's currently not a method for someone to swap a template directly with another templateType.

## How?
Add a new ReplaceTemplateButton component which opens a modal of available templates and lets a user pick the correct on.

## Testing Instructions
1. Open the Site Editor and go to the template inspector.
2. You should see a button that allows you to replace your current template
3. This button should open a modal allowing you to select a different template
4. When you select a different template, the contents of your old template should be replace with the one you selected and the entity should be dirty.

## Screenshots or screencast 



https://github.com/WordPress/gutenberg/assets/107534/391a3c24-4b5d-4754-b51b-4c99851097b2




This work is heavily based on https://github.com/WordPress/gutenberg/pull/51477
